usd/src/UsdTestUtils.hh

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef SDF_USD_TEST_UTILS_HH_
+#define SDF_USD_TEST_UTILS_HH_
+
+#include <gtest/gtest.h>
+#include <ignition/math/Angle.hh>
+#include <ignition/math/Pose3.hh>
+#include <pxr/base/gf/vec3d.h>
+#include <pxr/base/gf/vec3f.h>
+#include <pxr/base/tf/token.h>
+#include <pxr/usd/usd/prim.h>
+
+namespace usd
+{
+namespace testing
+{
+
+/// \brief Compare the pose of a USD prim to a desired pose
+/// \param[in] _usdPrim The USD prim
+/// \param[in] _targetPose The pose that _usdPrim should have
+void CheckPrimPose(const pxr::UsdPrim &_usdPrim,
```

Comment:
Is it possible that there will be no translate/rotate attributes if there are no relevant transforms?

Reply:
Can you explain/give an example of what you mean by "no relevant transforms"?

Reply:
I'm thinking in the case that e.g. all the translate are 0, there may not be any translate op, but now I see the code which generate the pose, the operators are always present regardless.